src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -213,6 +152,54 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    Profile WSLDistro{};
+    HANDLE readPipe;
+    HANDLE writePipe;
+    SECURITY_ATTRIBUTES sa{ sizeof(sa), nullptr, true };
+    if (CreatePipe(&readPipe, &writePipe, &sa, 0)) {
+        STARTUPINFO si{};
+        si.cb = sizeof(si);
+        si.dwFlags = STARTF_USESTDHANDLES;
+        si.hStdOutput = writePipe;
+        si.hStdError = writePipe;
+
+        PROCESS_INFORMATION pi{};
+        std::wstring command = L"wsl.exe --list";
+        HRESULT hr = CreateProcessW(nullptr, const_cast<LPWSTR>(command.c_str()), nullptr, nullptr,
+                                                        TRUE, CREATE_NO_WINDOW, nullptr, nullptr, &si, &pi);
+        if (SUCCEEDED(hr)) {
+            WaitForSingleObject(pi.hProcess, INFINITE);
+            DWORD exitCode;
+            if ((GetExitCodeProcess(pi.hProcess, &exitCode) == false) || (exitCode != 0)) {
+                hr = E_INVALIDARG;
+            }
+            if (SUCCEEDED(hr)) {
+                wchar_t buffer[500]; //arbitrary number.  Would like to evaluate other options for length of ouput

```

Comment:
Max distro name length is 256, plus another 10 for " (Default)", maybe round up to the next power of 2 to be safe (512).

Reply:
Where does the limit come from? How about using a defined value (e.g. UCHAR_MAX) instead of const value?

Reply:
@benhillis I'm using the buffer to grab the whole console output, so with my current code that could break some situations.  I'll see if I can figure out how to read a single line at a time from the pipe rather that pulling everything and then splitting the lines.

The other option I see is to have a huge buffer with `256 * MAX_DISTROS`, which is probably another fairly large number.

@Biswa96 `UCHAR_MAX `may be a little misleading (kind of like binarycrusader pointed out with my `NULL `vs `nullptr`).  Should I define a separate macro to define the maximum distribution name length for WSL distros?

Reply:
I've eliminated the buffer by figuring out how to convert the pipe into a stream using the following line:
`std::wifstream pipe{ _wfdopen( _open_osfhandle((intptr_t)readPipe, _O_WTEXT), L"r") };`